### ISSUE TITLE
cannot run docker-compose up --build

### ISSUE THREAD

**Issue by Fatima-yo** [gitcoinco/web#5549]
**Describe the bug**
While trying to get the local docker repo up and running, I get to the step when I must run " docker-compose up --build" and I get and error.

**To Reproduce**
1. follow the instructions here:
https://docs.gitcoin.co/mk_setup/
2. until you run the following:
docker-compose up --build

**Expected behavior**
The docker container should start up.

**Screenshots**
I get the following error displayed on the console:
ERROR: unsatisfiable constraints:
  so:libheif.so.1 (missing):
    required by: vips-8.8.3-r2[so:libheif.so.1]
                 vips-8.8.3-r2[so:libheif.so.1]
                 vips-8.8.3-r2[so:libheif.so.1]
ERROR: Service 'web' failed to build: The command '/bin/sh -c apk add --no-cache --update $PACKAGES &&     apk add --no-cache --update --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/ vips &&     apk add --no-cache --update --virtual .builder $BUILD_DEPS' returned a non-zero code: 2

**Desktop (please complete the following information):**
- OS: Ubuntu Studio 19
- Docker Version: docker-compose version 1.21.0
- Gitcoin Version: 1.0.0

**Comment by mul1sh** [comment]
Same here running into the same issue @thelostone-mc @danlipert

**Comment by Korridzy** [comment]
There is a good solution for this issue. The vips package will be installed from community repo and libheif from main.

I'm preparing a PR with this solution. @mul1sh, you can use it to unblock your work meanwhile.

```shell
# Install general dependencies.
RUN apk add --no-cache --update $PACKAGES && \
apk add --no-cache --update --repository http://dl-cdn.alpinelinux.org/alpine/edge/main/ --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/ vips && \
apk add --no-cache --update --virtual .builder $BUILD_DEPS
```

**Comment by Fatima-yo** [comment]
Thank you !
Waiting for it.

**Comment by mul1sh** [comment]
@Korridzy thanks, this seems to work though now it's hanging at the installing vips command i.e. `Installing vips (8.8.3-r2)` i've restarted the installation severally with no luck :(

**Comment by Korridzy** [comment]
Yes, it looks like problem is a bit more complex. Containers were created succesfully in my environment. But 'web', 'scheduler' and 'worker' now can't start with similar messages like:

```
OSError: cannot load library 'libvips.so.42': Error relocating /usr/lib/libheif.so.1: _ZNSt7__cxx1118basic_stringstreamIcSt11char_traitsIcESaIcEEC1Ev: symbol not found.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libvips.so.42'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/pyvips/__init__.py", line 19, in <module>
    import _libvips
ModuleNotFoundError: No module named '_libvips'
```
Investigation is in process

**Comment by octavioamu** [comment]
probably related https://github.com/alpinelinux/docker-alpine/issues/49

**Comment by octavioamu** [comment]
Our problem https://gitlab.alpinelinux.org/alpine/aports/issues/10962

**Comment by octavioamu** [comment]
the blame commit https://github.com/alpinelinux/aports/commit/2f6bc1cb9b406078ebc8e9621231a50159680dd9

**Comment by octavioamu** [comment]
ok seems this worked, https://github.com/gitcoinco/web/pull/5553 you can try that changes locally until we figure out a better solution

**Comment by mul1sh** [comment]
@octavioamu thanks for the follow-up but still no luck in getting this work even with the command updates. I'm now getting the following error
```sh
ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-2iyl7n63/secp256k1/setup.py'"'"'; __file__='"'"'/tmp/pip-install-2iyl7n63/secp256k1/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-63nn0nwo --python-tag cp37
       cwd: /tmp/pip-install-2iyl7n63/secp256k1/
  Complete output (210 lines):
...
  ERROR: Failed building wheel for secp256k1
```

**Comment by mul1sh** [comment]
@octavioamu thanks for the follow-up and the Dockerfile update, but still no luck in getting this work even with the command updates. I'm now getting the following error
```sh
ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-2iyl7n63/secp256k1/setup.py'"'"'; __file__='"'"'/tmp/pip-install-2iyl7n63/secp256k1/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-63nn0nwo --python-tag cp37
       cwd: /tmp/pip-install-2iyl7n63/secp256k1/
  Complete output (210 lines):
...
  ERROR: Failed building wheel for secp256k1
```

I think its because of the alpine update.

**Comment by octavioamu** [comment]
@mul1sh I think you tested #5554 not #5553

**Comment by mul1sh** [comment]
@octavioamu Yes I did because #5554 was merged into the repo whereas #5553 was closed. Anyway thanks for the heads up, let test again with #5553

**Comment by thelostone-mc** [comment]
sorry about that @mul1sh ! goofup at my end 

@Fatima-yo could you pull in master and try again ! Hopefully that should work

**Comment by Fatima-yo** [comment]
Yes, will do and let you know the results here.

**Comment by mul1sh** [comment]
@thelostone-mc no worries, thanks for the update 🙂 . I'll test later today then.

**Comment by thelostone-mc** [comment]
UPDATE: this still seems to be an issue ! mind giving us a day to figure it out ? 
I've been trying to build my container all day 😅

**Comment by Fatima-yo** [comment]
OK waiting for your ready-steady-go signal

**Comment by thelostone-mc** [comment]
@Fatima-yo + @mul1sh So I've got a working setup over at 
https://github.com/gitcoinco/web/pull/5558. This has been merged in so pull in the latest and rebuild the container.

Figuring out the breaking dependencies was tricky so we ended up shifting to ubuntu image as opposed to alpine

**Comment by Fatima-yo** [comment]
Great!
And funny also, as Ubuntu is my favorite, I run Xubuntu and Ubuntu Studio currently (plus Lubuntu a while ago and now LinuxLite in a small machine I use for YouTube only)

**Comment by thelostone-mc** [comment]
When you do get the time , could you pull in the latest master branch and rebuild the container.
If it's good -> I'll go ahead and close this issue 

Thanks for reporting it and being super patient ^_^

**Comment by Fatima-yo** [comment]
Yeah will do
Not sure if today (almost going to sleep and still got a few issues to close) but tomorrow early morning

**Comment by Fatima-yo** [comment]
Looks like it is working now :-D
![Screenshot_2019-11-29_16-51-21](https://user-images.githubusercontent.com/3955081/69889699-2d8e8080-12d1-11ea-9089-a41bb6b68376.png)
![Screenshot_2019-11-29_17-34-07](https://user-images.githubusercontent.com/3955081/69889700-2e271700-12d1-11ea-9e9b-c798a985cc9a.png)
![Screenshot_2019-11-29_17-45-15](https://user-images.githubusercontent.com/3955081/69889701-2e271700-12d1-11ea-8ba5-fd7ccf4587ec.png)
![Screenshot_2019-11-29_17-51-48](https://user-images.githubusercontent.com/3955081/69889702-2e271700-12d1-11ea-8b07-bb7c8bc89864.png)

**Comment by Fatima-yo** [comment]
@thelostone-mc kindly check my last comment and tell me if I should test anything else.
Looks as everything is good now.

**Comment by mul1sh** [comment]
No luck for me 😭 , it always exits with an error as shown below. And this after trying to build the image more than 5 times
<img width="972" alt="Screenshot 2019-11-30 at 5 21 58 PM" src="https://user-images.githubusercontent.com/2150634/69901703-02ba3000-1396-11ea-90ed-7169a6348025.png">

**Comment by Fatima-yo** [comment]
@mul1sh are you sure about having the latest docker version?
I installed it following these instructions:
https://docs.docker.com/install/linux/docker-ce/ubuntu/

**Comment by octavioamu** [comment]
@mul1sh be sure you run a fresh installation 
`docker-compose down -v; docker-compose up -d --build;`

**Comment by mul1sh** [comment]
@Fatima-yo  Yep i'm using the latest docker version 2.10.5

@octavioamu  Yes i'm aslo running a fresh installation, infact i'm purging the old containers at every retry and clearing my cache just in case, but no luck it always fails at the above line.

**Comment by octavioamu** [comment]
As I see on the error seems you are under some vpn or something on your network is blocking the connection. If you have a vpn try using it. Or not sure if your up is fixed, try to reload your modem to see if with another up works

**Comment by mul1sh** [comment]
@octavioamu ok let me try switching ISP's and see if it helps, thanks

**Comment by Fatima-yo** [comment]
@thelostone-mc kindly let me know when should I close this issue

**Comment by thelostone-mc** [comment]
@Fatima-yo SOLIDD :D  ❤️ 
Sorry I was with family and hence stayed away from github over the weekend 
We are good to close it 

@mul1sh I've gotten that error cause of network being blocked !
Use a VPN like @octavioamu and it should be a breeze

**Comment by Fatima-yo** [comment]
Closing issue as it was resolved

**Comment by Fatima-yo** [comment]
I am closing issue as it was resolved

**Comment by octavioamu** [comment]
@mul1sh were you able to solve your problem?

**Comment by mul1sh** [comment]
@octavioamu I was 🙂, thanks.
